microsoft/testsuites/osu/osusuite.py

```diff
@@ -0,0 +1,165 @@
+from lisa.operating_system import Posix
+from lisa.tools import Chmod, Lscpu, Make, Tar, Wget
+
+OSU_MPI_LOCATION = (
+    "https://mvapich.cse.ohio-state.edu/download/mvapich/mv2/mvapich2-2.3.7-1.tar.gz"
+)
+OSU_MPI_FILE_NAME = "mvapich2.tgz"
+
+OSU_BENCH_LOCATION = (
+    "http://mvapich.cse.ohio-state.edu/download/mvapich/osu-micro-benchmarks-6.2.tar.gz"
+)
+OSU_BENCH_FILE_NAME = "osu-bench.tgz"
+
+
+@TestSuiteMetadata(
+    area="hpc",
+    category="functional",
+    name="OSU Bench",
+    description="""
+    This test suite runs the OSU Micro-Benchmarks MPI test cases.
+    """,
+)
+class OSUTestSuite(TestSuite):
+    TIMEOUT = 200000
+
+    @TestCaseMetadata(
+        description="""
+            This test case runs GPU/CPU MPI latency.
+
+            Steps:
+            1. Install MVAPICH;
+            2. Install OSU Micro-Benchmarks;
+            3. Run GPU/CPU collective/latency tests in a single node.
+
+        """,
+        timeout=TIMEOUT,
+        requirement=simple_requirement(
+            supported_features=[GpuEnabled(), SerialConsole],
+        ),
+        priority=1,
+    )
+    def run_osu_bench(self, node: Node, log: Logger, log_path: Path) -> None:
+
+        _install_osu_mpi(node)
+        path = _install_osu_bench(node)
+
+        gpu_count = node.capability.gpu_count
+        gpu = node.features[Gpu]
+        expected_count = gpu.get_gpu_count_with_lspci()
+        assert_that(gpu_count).described_as(
+            "Expected device count didn't match actual device count from lspci!"
+        ).is_equal_to(expected_count)
+
+        nproc = node.tools[Lscpu].get_core_count()
+        tests = [
+            "allgather",
+            "allreduce",
+            "alltoall",
+            "bcast",
+            "gather",
+            "reduce_scatter",
+            "reduce",
+            "scatter",
+        ]
+        for test in tests:
+            node.execute(
+                f"mpirun -np {nproc} python3.8 {path}/python/run.py"
+                f" --benchmark {test} --min 128 --max 10000000 >> "
+                f"{path}/osu_bench.log 2>&1",
+                expected_exit_code=0,
+                expected_exit_code_failure_message=(
+                    f"failed to run mpirun {test} benchmark"
+                ),
+                timeout=1200,
+                shell=True,
+                sudo=True,
+            )
+        log.info("OSU MPI tests finish!")
+        osu_log_path = str(node.get_pure_path(f"{path}/osu_bench.log"))
+        node.tools[Chmod].chmod(osu_log_path, "a+rwX", sudo=True)
+        node.shell.copy_back(
+            node.get_pure_path(osu_log_path), PurePath(log_path) / "osu_bench.log"
+        )
+
+
+def _install_osu_mpi(node: Node) -> None:
+    assert isinstance(node.os, Posix)
+    node.os.install_packages("libibverbs-dev gfortran bison")
+    wget = node.tools[Wget]
+    tar = node.tools[Tar]
+
+    path = wget.get_tool_path(use_global=True)
+    extracted_path = tar.get_tool_path(use_global=True)
```

Comment:
@LiliDeng This will extract folder in a system directory. Was this change intended?

Reply:
I just used previous code and didn't notice this.

Reply:
No problem. I thought these were intentional. @darkwhite29 Do you have a copy of changes we worked on prior to Lili's changes?